src/pveclib/vec_int512_ppc.h

```diff
@@ -840,134 +625,100 @@ vec_mul128x128_PWR9 (vui128_t m1l, vui128_t m2l)
 #include <pveclib/vec_int128_ppc.h>
 #include <pveclib/vec_int512_ppc.h>
-
+ ...
 // vec_mul128x128_inline is defined in vec_int512_ppc.h
 __VEC_U_256
 __VEC_PWR_IMP (vec_mul128x128) (vui128_t m1l, vui128_t m2l)
 {
   return vec_mul128x128_inline (m1l, m2l);
 }
  * \endcode
- * This simple strategy allows the collection of the larger function
- * implementations into a single source file and build object files
- * for multiple platform targets. For example collect all the multiple
- * precision quadword implementations into a source file named
- * <B>vec_int512_runtime.c</B>.
- *
- * This source file can be compiled multiple times for different
- * platform targets. The resulting object files have unique function
- * symbols due to the platform specific suffix provided by the
- * __VEC_PWR_IMP() macro.
- * There are a number of build strategies for this.
- *
- * For example, create a small source file named <B>vec_runtime_PWR8.c</B>that starts with the
- * target pragma and includes the multi-platform source file.
- * \code
- //
- //  \file  vec_runtime_PWR8.c
- //
-
-#pragma GCC target ("cpu=power8")
-
-#include "vec_int512_runtime.c"
- * \endcode
- * Similarly for <B>vec_runtime_PWR7.c</B>, <B>vec_runtime_PWR9.c</B>.
- *
- * \note Additional runtime source files can be included as needed.
- * Other multiple precision functions supporting BCD and BCD <-> binary
- * conversions are likely candidates.
- *
- * Then include these files in the Makefile.am list of sources to build.
- * \code
-libpvec_la_SOURCES = vec_runtime_PWR9.c \
-	vec_runtime_PWR8.c \
-	vec_runtime_PWR7.c
- * \endcode
- *
- * Alternatively use Makefile.am rules to specify different -mcpu= compile options.
- * This simplifies the platform source files too:
- * \code
- //
- //  \file  vec_runtime_PWR8.c
- //
-
-#include "vec_int512_runtime.c"
- * \endcode
- * Then add the -mcpu compile option to CFLAGS
- * \code
-CFLAGS-vec_runtime_PWR7.c += -mcpu=power7
-CFLAGS-vec_runtime_PWR8.c += -mcpu=power8
-CFLAGS-vec_runtime_PWR9.c += -mcpu=power9
- * \endcode
- *
- * \todo Is there a way for automake to compile vec_int512_runtime.c
- * with -mcpu=power7 with -o vec_runtime_PWR7.o.
- * And similarly for PWR8/PWR9.
- *
- * \subsection i512_libary_issues_0_0_1 Calling Multi-platform functions
- *
- * The next step is to provide mechanisms for applications to call
- * these functions via static or dynamic linkage.
- * For static linkage the application needs to reference a specific
- * platform variant of the functions name.
- * For dynamic linkage we will use <B>STT_GNU_IFUNC</B> symbol resolution
- * (a symbol type extension to the ELF standard).
- *
- * \subsubsection i512_libary_issues_0_0_1_1 Static linkage to platform specific functions
+ * This ensures that target specific runtime
+ * implementations have unique function symbols. This is important to
+ * avoid linker errors (due to duplicate symbol names).
+ *
+ * \note Each runtime operation will have 2 or 3 target qualified
```

Comment:
You qualify with "currently" below. Maybe that should be moved up here.  POWER10 is coming.

Reply:
How about:
 * As new POWER processors are released, additional targets will be added.

Reply:
Also using "target" here instead of "platform". Target aligns with the GCC compiler terminology while platform is from the AuxV PLATFORM  entry. Platform is easy to confuse with PPC vs X86.